server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/BroadcastTest.java

```diff
@@ -0,0 +1,132 @@
+        ConnectionFactory connectionFactory = new ConnectionFactory();
+        connectionFactory.setHost(rabbitMQ.getHostIp());
+        connectionFactory.setPort(rabbitMQ.getPort());
+        connectionFactory.setUsername(rabbitMQ.getUsername());
+        connectionFactory.setPassword(rabbitMQ.getPassword());
+        return connectionFactory;
+    }
+
+    @Test
+    // In the following case, each consumer will receive the messages produced by the
+    // producer
+
+    // To do so, each consumer will bind it's queue to the producer exchange.
+    public void rabbitMQShouldSupportTheBroadcastCase() throws Exception {
+        ConcurrentLinkedQueue<Integer> results2 = new ConcurrentLinkedQueue<>();
+        ConcurrentLinkedQueue<Integer> results3 = new ConcurrentLinkedQueue<>();
+        ConcurrentLinkedQueue<Integer> results4 = new ConcurrentLinkedQueue<>();
+
+        try (Connection connection1 = connectionFactory1.newConnection();
+             Channel publisherChannel = connection1.createChannel();
+            Connection connection2 = connectionFactory2.newConnection();
+             Channel subscriberChannel2 = connection2.createChannel();
+            Connection connection3 = connectionFactory3.newConnection();
+             Channel subscriberChannel3 = connection3.createChannel();
+            Connection connection4 = connectionFactory4.newConnection();
+             Channel subscriberChannel4 = connection4.createChannel()) {
+
+            // Declare the a single exchange and three queues attached to it.
+            publisherChannel.exchangeDeclare(EXCHANGE_NAME, "direct", DURABLE);
+
+            String queue2 = subscriberChannel2.queueDeclare().getQueue();
+            subscriberChannel2.queueBind(queue2, EXCHANGE_NAME, ROUTING_KEY);
+            String queue3 = subscriberChannel3.queueDeclare().getQueue();
+            subscriberChannel3.queueBind(queue3, EXCHANGE_NAME, ROUTING_KEY);
+            String queue4 = subscriberChannel4.queueDeclare().getQueue();
+            subscriberChannel4.queueBind(queue4, EXCHANGE_NAME, ROUTING_KEY);
+
+            subscriberChannel2.basicConsume(queue2, storeInResultCallBack(subscriberChannel2, results2));
+            subscriberChannel3.basicConsume(queue3, storeInResultCallBack(subscriberChannel3, results3));
+            subscriberChannel4.basicConsume(queue4, storeInResultCallBack(subscriberChannel4, results4));
+
+            // the publisher will produce 10 messages
+            for (int i = 0; i < 10; i++) {
+                publisherChannel.basicPublish(EXCHANGE_NAME, ROUTING_KEY, NO_PROPERTIES,
+                    String.valueOf(i).getBytes(StandardCharsets.UTF_8));
+            }
+
+            Thread.sleep(ONE_MINUTE);
+
+            // Check every subscriber have receive all the messages.
+            ImmutableList<Integer> expectedResult = IntStream.range(0, 10).boxed().collect(Guavate.toImmutableList());
```

Comment:
you could use this list for the previous loop, it would allow to get rid of the old school for loop